src/instructlab/configuration.py

```diff
@@ -339,25 +357,74 @@ class _generate(BaseModel):
     seed_file: StrictStr = Field(default_factory=lambda: DEFAULTS.SEED_FILE)
     gpus: Optional[int] = None
 
+    @model_validator(mode="before")
+    def fill_defaults(
+        cls, values: typing.Dict[str, typing.Any]
+    ) -> typing.Dict[str, typing.Any]:
+        defaults = {
+            "model": DEFAULTS.DEFAULT_MODEL,
+            "taxonomy_path": DEFAULTS.TAXONOMY_DIR,
+            "taxonomy_base": DEFAULTS.TAXONOMY_BASE,
+        }
+
+        return finish_cfg_section(defaults, values)
+
 
 class _mmlu(BaseModel):
     few_shots: int
     batch_size: str
 
+    @model_validator(mode="before")
+    def fill_defaults(
+        cls, values: typing.Dict[str, typing.Any]
+    ) -> typing.Dict[str, typing.Any]:
+        defaults = {"few_shots": 16, "batch_size": "auto"}
+        return finish_cfg_section(defaults, values)
+
 
 class _mtbench(BaseModel):
     judge_model: str
     output_dir: str
     max_workers: int
 
+    @model_validator(mode="before")
+    def fill_defaults(
+        cls, values: typing.Dict[str, typing.Any]
+    ) -> typing.Dict[str, typing.Any]:
+        defaults = {
+            "judge_model": DEFAULTS.JUDGE_MODEL_MT,
+            "output_dir": DEFAULTS.EVAL_DATA_DIR,
+            "max_workers": 16,
+        }
+        return finish_cfg_section(defaults, values)
+
 
 class _mtbenchbranch(BaseModel):
     taxonomy_path: str
 
+    @model_validator(mode="before")
+    def fill_defaults(
+        cls, values: typing.Dict[str, typing.Any]
+    ) -> typing.Dict[str, typing.Any]:
+        defaults = {
+            "judge_model": DEFAULTS.JUDGE_MODEL_MT,
```

Comment:
I think there's a bug here: the defaults dict doesn't match the class. Not that I think this code should stay, but in case we want to stick to this version of the implementation, this will have to be fixed.

Reply:
fixed!